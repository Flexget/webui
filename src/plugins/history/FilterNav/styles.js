import styled from '@emotion/styled';
import { css } from '@emotion/core';

import { Icon } from '@material-ui/core';

export const RotatingIcon = styled(Icon)`
  transition: ${({ theme }) => theme.transitions.create()};
  transform: ${({ rotate }) => rotate && 'rotate(180deg)'};
`;

export const Padded = styled.div`
  padding: 0.5rem 2rem 0.5rem;
`;

export const textField = css`
  width: 7.5rem;
`;
