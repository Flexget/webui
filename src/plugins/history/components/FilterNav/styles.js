import styled from 'react-emotion';
import Icon from '@material-ui/core/Icon';
import TextField from '@material-ui/core/TextField';
import theme from 'theme';

export const RotatingIcon = styled(Icon)`
  transition: ${theme.transitions.create()};
  transform: ${({ rotate }) => rotate && 'rotate(180deg)'};
`;

export const PaddedTextField = styled(TextField)`
  margin: 0.5rem 2rem 0.5rem;
`;
