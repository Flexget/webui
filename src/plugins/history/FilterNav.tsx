import React, { FC, useEffect } from 'react';
import { Toolbar, Tooltip } from '@material-ui/core';
import { css } from '@emotion/core';
import SelectField from 'common/inputs/formik/SelectField';
import { Spacer } from 'common/styles';
import { useFormikContext } from 'formik';
import { useDebounce } from 'utils/hooks';
import DirectionButton from 'common/inputs/formik/DirectionButton';
import { TaskContainer } from 'core/tasks/hooks';
import { useContainer } from 'unstated-next';
import { GroupByFields, GetHistoryOptions, SortByFields } from './types';

const fieldWrapper = css`
  padding: 0.5rem 2rem 0.5rem;
`;

const textField = css`
  width: 7.5rem;
`;

const selectField = css`
  min-width: 20rem;
`;
const groupByFields = [
  {
    value: GroupByFields.Task,
    label: 'Task',
  },
  {
    value: GroupByFields.Time,
    label: 'Time',
  },
];

const sortByFields = [
  {
    value: SortByFields.Details,
    label: 'Details',
  },
  {
    value: SortByFields.Filename,
    label: 'Filename',
  },
  {
    value: SortByFields.Id,
    label: 'ID',
  },
  {
    value: SortByFields.Task,
    label: 'Task',
  },
  {
    value: SortByFields.Time,
    label: 'Time',
  },
  {
    value: SortByFields.Title,
    label: 'Title',
  },
  {
    value: SortByFields.Url,
    label: 'URL',
  },
];

const FilterNav: FC = () => {
  const { values, submitForm } = useFormikContext<GetHistoryOptions>();

  const debouncedValues = useDebounce(values);
  const { tasks } = useContainer(TaskContainer);

  useEffect(() => {
    submitForm();
  }, [...Object.values(debouncedValues), submitForm]);

  return (
    <Toolbar>
      <div css={fieldWrapper}>
        <SelectField
          css={selectField}
          autoFocus
          id="task"
          label="Task Name"
          name="task"
          options={tasks.map(({ name }) => ({ value: name, label: name }))}
        />
      </div>
      <Spacer />
      <div css={fieldWrapper}>
        <SelectField id="sort" name="sort" label="Sort By" options={sortByFields} css={textField} />
      </div>
      <div>
        <SelectField
          id="grouping"
          label="Group By"
          name="grouping"
          options={groupByFields}
          css={textField}
        />
      </div>
      <Tooltip title="Toggle Direction">
        <DirectionButton color="inherit" name="order" id="order" aria-label="Toggle Direction" />
      </Tooltip>
    </Toolbar>
  );
};

export default FilterNav;
