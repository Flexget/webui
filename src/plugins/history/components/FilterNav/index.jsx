import React, { Component } from 'react';
import PropTypes from 'prop-types';
import IconButton from 'material-ui/IconButton';
import { MenuItem } from 'material-ui/Menu';
import SecondaryNav from 'components/SecondaryNav';
import { Spacer } from 'components/styles';
import { RotatingIcon, PaddedTextField } from './styles';

const ENTER_KEY = 13;
const groupByFields = [
  {
    value: 'task',
    label: 'Task',
  },
  {
    value: 'time',
    label: 'Time',
  },
];

const sortByFields = [
  {
    value: 'details',
    label: 'Details',
  },
  {
    value: 'filename',
    label: 'Filename',
  },
  {
    value: 'id',
    label: 'ID',
  },
  {
    value: 'task',
    label: 'Task',
  },
  {
    value: 'time',
    label: 'Time',
  },
  {
    value: 'title',
    label: 'Title',
  },
  {
    value: 'url',
    label: 'URL',
  },
];

class FilterNav extends Component {
  static propTypes = {
    order: PropTypes.oneOf(['asc', 'desc']).isRequired,
    grouping: PropTypes.oneOf(groupByFields.map(({ value }) => value)).isRequired,
    sort: PropTypes.oneOf(sortByFields.map(({ value }) => value)).isRequired,
    toggleOrder: PropTypes.func.isRequired,
    handleChange: PropTypes.func.isRequired,
  };

  state = {
    task: '',
  }

  handleChange = (event) => {
    this.setState({ task: event.target.value });
  }

  handleKeyPress = (event) => {
    const { handleChange } = this.props;
    if (event.which === ENTER_KEY) {
      handleChange('task')(event);
    }
  }

  render() {
    const {
      order,
      grouping,
      sort,
      toggleOrder,
      handleChange,
    } = this.props;

    const {
      task,
    } = this.state;

    return (
      <SecondaryNav>
        <PaddedTextField
          label="Filter By Task"
          id="task-filter"
          placeholder="Task Name"
          value={task}
          onChange={this.handleChange}
          onKeyPress={this.handleKeyPress}
        />
        <Spacer />
        <PaddedTextField
          id="sort-by"
          select
          label="Sort By"
          value={sort}
          onChange={handleChange('sort')}
        >
          {sortByFields.map(({ value, label }) => (
            <MenuItem value={value} key={value}>{label}</MenuItem>
          ))}
        </PaddedTextField>
        <PaddedTextField
          id="group-by"
          label="Group By"
          select
          value={grouping}
          onChange={handleChange('grouping')}
        >
          {groupByFields.map(({ value, label }) => (
            <MenuItem value={value} key={value}>{label}</MenuItem>
          ))}
        </PaddedTextField>
        <IconButton color="inherit" onClick={toggleOrder}>
          <RotatingIcon rotate={order === 'desc'} className="fa fa-chevron-up" />
        </IconButton>
      </SecondaryNav>
    );
  }
}

export default FilterNav;
