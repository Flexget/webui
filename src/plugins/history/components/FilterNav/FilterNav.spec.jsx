import React from 'react';
import renderer from 'react-test-renderer';
import FilterNav from 'plugins/history/components/FilterNav';

describe('plugins/history/components/FilterNav', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <FilterNav
        handleChange={jest.fn()}
        toggleOrder={jest.fn()}
        grouping="time"
        sort="time"
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
