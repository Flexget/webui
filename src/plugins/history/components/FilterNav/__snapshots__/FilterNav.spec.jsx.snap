// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plugins/history/components/FilterNav renders correctly 1`] = `
.glamor-6 {
  position: static;
}

.glamor-5 {
  background-color: #455a64;
  min-height: 5rem;
  color: #fff;
}

.glamor-0 {
  margin: 0.5rem 2rem 0.5rem;
}

.glamor-1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.glamor-4 {
  -webkit-transition: all 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: all 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
}

<header
  className="MuiPaper-root-9 MuiPaper-shadow4-15 MuiAppBar-root-1 MuiAppBar-positionFixed-2 MuiAppBar-colorPrimary-7 mui-fixed glamor-6 e1j0gihi0"
>
  <div
    className="MuiToolbar-root-36 MuiToolbar-gutters-37 glamor-5 e1j0gihi1"
  >
    <div
      className="MuiFormControl-root-38 glamor-0 ecdi3071"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyPress={[Function]}
    >
      <label
        className="MuiFormLabel-root-48 MuiInputLabel-root-42 MuiInputLabel-formControl-43 MuiInputLabel-animated-46"
        data-shrink={false}
        htmlFor="task-filter"
      >
        Filter By Task
      </label>
      <div
        className="MuiInput-root-52 MuiInput-formControl-53 MuiInput-inkbar-54 MuiInput-underline-58"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <input
          aria-invalid={false}
          aria-required={false}
          className="MuiInput-input-61"
          disabled={false}
          id="task-filter"
          onChange={[Function]}
          placeholder="Task Name"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="glamor-1 e1vib4761"
    />
    <div
      className="MuiFormControl-root-38 glamor-0 ecdi3071"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <label
        className="MuiFormLabel-root-48 MuiInputLabel-root-42 MuiInputLabel-formControl-43 MuiInputLabel-animated-46 MuiInputLabel-shrink-45"
        data-shrink={true}
        htmlFor="sort-by"
      >
        Sort By
      </label>
      <div
        className="MuiInput-root-52 MuiInput-formControl-53 MuiInput-inkbar-54 MuiInput-underline-58"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <div
          className="MuiSelect-root-67"
        >
          <div
            aria-haspopup="true"
            aria-owns={null}
            aria-pressed="false"
            className="MuiSelect-select-68 MuiSelect-selectMenu-69 MuiInput-input-61"
            onBlur={[Function]}
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            Time
          </div>
          <input
            aria-invalid={false}
            aria-required={false}
            id="sort-by"
            type="hidden"
            value="time"
          />
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-72 MuiSelect-icon-71"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      className="MuiFormControl-root-38 glamor-0 ecdi3071"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <label
        className="MuiFormLabel-root-48 MuiInputLabel-root-42 MuiInputLabel-formControl-43 MuiInputLabel-animated-46 MuiInputLabel-shrink-45"
        data-shrink={true}
        htmlFor="group-by"
      >
        Group By
      </label>
      <div
        className="MuiInput-root-52 MuiInput-formControl-53 MuiInput-inkbar-54 MuiInput-underline-58"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <div
          className="MuiSelect-root-67"
        >
          <div
            aria-haspopup="true"
            aria-owns={null}
            aria-pressed="false"
            className="MuiSelect-select-68 MuiSelect-selectMenu-69 MuiInput-input-61"
            onBlur={[Function]}
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            Time
          </div>
          <input
            aria-invalid={false}
            aria-required={false}
            id="group-by"
            type="hidden"
            value="time"
          />
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-72 MuiSelect-icon-71"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
    <button
      className="MuiButtonBase-root-89 MuiIconButton-root-83 MuiIconButton-colorInherit-84"
      disabled={false}
      onBlur={[Function]}
      onClick={[MockFunction]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label-88"
      >
        <span
          aria-hidden="true"
          className="material-icons MuiIcon-root-91 fa fa-chevron-up glamor-4 ecdi3070"
          rotate={false}
        />
      </span>
      <span
        className="MuiTouchRipple-root-98"
      />
    </button>
  </div>
</header>
`;
